operator/controllers/seldondeployment_controller.go

```diff
@@ -866,6 +866,39 @@ func createContainerService(deploy *appsv1.Deployment,
 	return svc
 }
 
+// compareServices ignores those fields set by k8s by default.
+// Specifically, the fields to be compared are:
+//   - Annotations
+//   - Labels
+//   - Spec.Ports
+//   - Spec.Type
+//   - Spec.Selector
+//   - Spec.SessionAffinity
+func compareServices(desiredSvc, found *corev1.Service) bool {
+	return equality.Semantic.DeepEqual(desiredSvc.Annotations, found.Annotations) &&
+		equality.Semantic.DeepEqual(desiredSvc.Labels, found.Labels) &&
+		equality.Semantic.DeepEqual(desiredSvc.Spec.Ports, found.Spec.Ports) &&
+		equality.Semantic.DeepEqual(desiredSvc.Spec.Selector, found.Spec.Selector) &&
+		equality.Semantic.DeepEqual(desiredSvc.Spec.Type, found.Spec.Type) &&
+		equality.Semantic.DeepEqual(desiredSvc.Spec.SessionAffinity, found.Spec.SessionAffinity)
```

Comment:
There are several other fields that can be set in the spec by the user: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#servicespec-v1-core
Have you validated none of these is applicable? Some have defaults and are usually not set so I suppose one would need to be careful if as we are doing now we are not copying the spec back.

Reply:
@cliveseldon, Thanks for you quick review. Do you mean in addition to the defaults set by Kubernetes, we also need to consider that the users may (intentionally or unintentionally) modify the services directly?

Reply:
Actually, as a user has very little control from  SeldonDeployment on the created service we can ignore fields like `externalIPs` etc as a user can't control these from Seldon.

Reply:
@cliveseldon. If we can ignore the impact of users, can I say that the change of svc fields are controlled by those defaults by Kubernetes and seldon operator? All the fields I selected to comapre/update are extracted from _createContainerServices_ and _ceratePredictorServices_ which are required by seldon operator side. I think keeping the defaults set by Kubernetes shouldn't bring side effects. Could you please give specific situations that we have to take other fields into account?

Reply:
No I think that is fine if we have covered all the items in the Spec that we control

Reply:
I think its ok as is